File: 22101528_Azimli_Maryam.kt
fun main() {
    //initialize
    val animal = mutableMapOf<String, Any>(
    "type" to "dog",
    "breed" to "husky",
    "age" to "4"
    )

    //Get the value for a given key
    println(animal["type"])

    //Add a new element
    animal["owner"] = "Maryam"
    println(animal["owner"])

    //remove an element
    animal.remove("age")
    println(animal["age"])

    //change value
    animal["owner"]="Katy"
    println(animal["owner"])

    //check whether the key exists or not
    if (animal.containsKey("breed")) {
        println("It exists")
    } else {
        println("It doesnt exist.")
    }

    //check if the value exists
    if (animal.containsValue("cat")) {
        println("It exists")
    } else {
        println("It doesnt exist.")
    }

    //loop through
    for ((key, value) in animal) {
        println("$key: $value")
    }
}

File: 22101528_Azimli_Maryam.rb
#initialize
animal = Hash.new
animal["type"] = "dog"
animal["breed"] = "husky"
animal["age"] = 4

#print
puts(animal["type"])

#add a new element
animal["owner"] = "Maryam"
puts(animal["owner"])

#remove an element
animal.delete("owner")
puts(animal["owner"])

#modify an element
animal["breed"] = "chihuahua"
puts(animal["breed"])

#search for key
if animal.key?("breed")
  puts ("key exists.")
else
  puts ("key doesnt exist")
end

#search for value
if animal.value?("husky")
  puts "value exists"
else
  puts "value does not exist"
end

#loop through
animal.each do |key, value|
  puts "#{key}: #{value}"
end